File: code/communication.py
import sys
# sys.path.append(r"C:\Python27\lib")
sys.path.append(r"C:\Python27\Lib\site-packages")
# import clr
import logging
import time
# import MissionPlanner

import socket

from socketIO_client import SocketIO, LoggingNamespace

logging.getLogger('socketIO-client').setLevel(logging.DEBUG)

HOST = 'localhost'
PORT = 50001
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)


# clr.AddReference("MissionPlanner.Utilities")
# print "Starting Mission"
# Script.ChangeMode("Guided")
# print "Guided Mode"
print "Connecting to localhost"

# Listen to online web server
with SocketIO('localHost', 8080, LoggingNamespace) as socketIO:

	# On connection
	def on_connect(*args):
		s.connect((HOST, PORT))
		print "Connected! ", args

	# Receive coordinates from web app
	def on_send(*args):
		print "Coordinates obtained: ", args[0]['lat'], args[0]['lng']
		s.sendall(str(args[0]['lat']) + "," + str(args[0]['lng']))

	def on_disconnect(*args):
		s.close()

	def on_request(*args):
		print "Requesting location:"
		s.sendall("request")
		data = s.recv(1024)
		print "Location received:", data
		socketIO.emit("location", data)



	socketIO.on('connect', on_connect)
	socketIO.on('send', on_send)
	socketIO.on('disconnect', on_disconnect)
	socketIO.on('request', on_request)
	socketIO.wait()

s.close()

File: code/onlineWebApp/communication.js
var express = require('express');
var app = express();
var server = require('http').Server(app);
var io = require('socket.io')(server);

app.use(express.static('.'));

app.get('/', function(req, res) {
	res.sendFile(__dirname + '/index.html');
});

io.on('connection', function(socket) {
	console.log('User connected');


	socket.on('disconnect', function() {
		console.log('user disconnected');
	});

	// Send the coordinates from web app to MissionPlanner
	socket.on('send', function(msg) {
		console.log('Received coordinates: ' + msg.lat + ", " + msg.lng);
		io.emit('send', msg);
	});

	socket.on('location', function(msg) {
		io.emit('location', msg);
	});
});

function requestPosition() {
	console.log('Requesting location.');
	io.sockets.emit('request', 'request')
}

setInterval(requestPosition, 5000);

server.listen(8080, function(){
	var host = server.address().address;
	var port = server.address().port;
	console.log('listening on ' + host + ":" + port);
});